lib/libbackscrub.h

```diff
@@ -0,0 +1,54 @@
+/* This is licenced software, @see LICENSE file.
+ * Authors - @see AUTHORS file.
+==============================================================================*/
+
+// tested against tensorflow lite v2.4.1 (static library)
+#ifndef _LIBBACKSCRUB_H
+#define _LIBBACKSCRUB_H
+
+// for cv::Mat and related types
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <stdarg.h>
+
+#define TFLITE_MINIMAL_CHECK(x)                              \
+  if (!(x)) {                                                \
+	fprintf(stderr, "Error at %s:%d\n", __FILE__, __LINE__); \
+	exit(1);                                                 \
+  }
+
+// Shared state structure between caller and libbackscrub
+// backscrub_ctx MUST be zero'ed by caller before first use.
+typedef struct {
+	// Required to be set before calling init_tensorflow
+	// XXX:TODO: poss should be formal params - PAA
+	const char *modelname;
+	size_t threads;
+	size_t width;
+	size_t height;
+	int debug;
+	// Optional callbacks with context between processing steps
+	void (*ondebug)(void *ctx, const char *fmt, va_list ap);
+	void (*onprep)(void *ctx);
+	void (*oninfer)(void *ctx);
+	void (*onmask)(void *ctx);
+	void *caller_ctx;
+	// Used by libbackscrub / callbacks (eg: adjusting blur size)
+	// XXX:TODO: probably too much coupling - PAA
+	cv::Mat input;
+	cv::Mat output;
+	cv::Rect roidim;
+	cv::Mat mask;
+	cv::Mat mroi;
+	cv::Mat raw;
+	cv::Mat ofinal;
+	cv::Size blur;
```

Comment:
Possibly avoid C++ here, to allow the header to be a plain C interface.
Also should include an `extern "C" {}` block around everything in here. Otherwise function names of the library will be mangled and thus interop with other languages will be a hassle.

Reply:
I started on that route, then decided to fix one thing at a time, so yep, C compatible API, which will be much easier when all this implementation detail is hidden away in the library, not shared with callers..

Reply:
All gone, but we may have a problem trying to expose a plain C API:
https://stackoverflow.com/questions/48365991/is-there-a-working-c-interface-for-opencv-3-x
looks like OpenCV no longer has a C API we can use to transport image data...

Reply:
I think we can do a somewhat hybrid API: Most functions can be exposed as wrappers with C compatibility and some direct C++ API when C++ is supported. In the worst case this just means, that some C++ API may be hidden for plain C usage, which is not great, but allows for easy fallback in such usecases. But mor on this in a proper issue.

Reply:
Yep, let's take this to a new issue :)